x-pack/platform/plugins/shared/stack_connectors/public/connector_types/email/email_connector.tsx

```diff
@@ -120,6 +122,26 @@ export const EmailActionConnectorFields: React.FunctionComponent<ActionConnector
   const disableServiceConfig = shouldDisableEmailConfiguration(service);
   const { isLoading, getEmailServiceConfig } = useEmailConfig({ http, toasts });
 
+  useMount(() => {
```

Comment:
I am a bit confused why we want this extra API call. Should the `useEffect` achieve the same? The `service` is either what the user selected or what the connector is already configured. If the service is not part of the available services, and the connector is configured with such a service, we should probably not set any value or deselect the value. I think it is related to a bug I posted in my second review.

Reply:
The useEffect is triggered every time the user clicks on the `select service` component. I added useMount to automatically load the data linked to a service when there is only one service available. For example, if we just have `amazon-ses`, it will select that service and load the host, port and issecure fields automatically when rendering the form. Related to the bug you found, it shouldn't trigger if it's an already existing connector that has a service selected. Still useful for new connectors when only one service is enabled, though

Reply:
It wasn't possible, at least not in a simple manner. Instead, the dropdown will have as options the enabled services. On update, it will render the enabled services and the currently stored service. If the user just updates the name, the BE will throw until the service is updated to an enabled option. The downside is that if only one service is enabled, we'll show a dropdown with just one option and the user will have to manually select that only option.